framework/db/Schema.php

```diff
@@ -399,6 +399,33 @@ public function setTransactionIsolationLevel($level)
     }
 
     /**
+     * Executes the INSERT command, returning primary key values.
+     * @param string $table the table that new rows will be inserted into.
+     * @param array $columns the column data (name => value) to be inserted into the table.
+     * @return array primary key values or false if the command fails
+     * @since 2.0.4
+     */
+    public function insert($table, $columns)
+    {
+        $command = $this->db->createCommand()->insert($table, $columns);
+        if (!$command->execute()) {
+            return false;
+        }
+        $tableSchema = $this->getTableSchema($table);
+        if ($tableSchema->sequenceName === null) {
+            return [];
```

Comment:
I agree, when there's no `sequenceName` it's probable that `$columns` contain new PK values anyway.

This is also related to my comment about the null check in AR, see above.